x/gov/client/cli/tx.go

```diff
@@ -0,0 +1,237 @@
+package cli
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"strconv"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/wire"
+	authcmd "github.com/cosmos/cosmos-sdk/x/auth/client/cli"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	"github.com/pkg/errors"
+)
+
+const (
+	flagProposalID   = "proposalID"
+	flagTitle        = "title"
+	flagDescription  = "description"
+	flagProposalType = "type"
+	flagDeposit      = "deposit"
+	flagProposer     = "proposer"
+	flagDepositer    = "depositer"
+	flagVoter        = "voter"
+	flagOption       = "option"
+)
+
+// submit a proposal tx
+func GetCmdSubmitProposal(cdc *wire.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "submitproposal",
+		Short: "Submit a proposal along with an initial deposit",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			title := viper.GetString(flagTitle)
+			description := viper.GetString(flagDescription)
+			proposalType := viper.GetString(flagProposalType)
+			initialDeposit := viper.GetString(flagDeposit)
```

Comment:
We should do some checks in the CLI on the content of theses viper inputs - aka fail if they're blank I reckon

Reply:
maybe some of these basic checks should be added to the messages `ValidateBasic` and then we can just call validate basic in the CLI here

Reply:
`ValidateBasic()` is a good idea, I think we can simulate the whole tx in most cases actually - https://github.com/cosmos/cosmos-sdk/issues/1246 - does not need to happen in this PR.